src/poke_env/environment/pokemon.py

```diff
@@ -654,9 +668,7 @@ def base_species(self) -> str:
         :rtype: str
         """
         dex_entry = self._data.pokedex[self._species]
-        if "baseSpecies" in dex_entry:
-            return to_id_str(dex_entry["baseSpecies"])
-        return self._species
+        return to_id_str(dex_entry["baseSpecies"])
```

Comment:
is this always defined? and why is this change in this PR?

Reply:
Ah, I'm sorry! This isn't strictly necessary for the PR -- would you like me to remove it?

For context, I think you already ensured that each species has a `baseSpecies` in `GenData` [here](https://github.com/hsahovic/poke-env/blob/a2cd6e8e5c88c0fac5f3f61d67d150af8a1253e4/src/poke_env/data/gen_data.py#L66) and so these lines were duplicative; I just cleaned this up for simplicity because it was hard for me to track down what baseSpecies actually was when I was doing this work

Reply:
Ah, i see. Ok, this makes sense.